server/svix-server/src/core/security.rs

```diff
@@ -110,14 +239,47 @@ where
         .one(db)
         .await?
         .ok_or_else(|| HttpError::not_found(None, None))?;
+
+        if let Some(permitted_app_id) = &permissions.app_id {
+            if permitted_app_id != &app.id {
+                return Err(HttpError::unauthorized(
+                    None,
+                    Some(
+                        "You are only permitted to perform operations under the Application type"
+                            .to_owned(),
+                    ),
+                )
```

Comment:
The error we currently return is 404, which makes sense as the app doesn't exist (for all this auth token knows).
We can however return a 403 like above. Either works for me, though not this.